src/components/LoginForm.component.tsx

```diff
@@ -0,0 +1,124 @@
+import { useLazyQuery } from "@apollo/client";
+import React, { useContext, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import styled from "styled-components";
+
+import { UserContext } from "../contexts/UserContext";
+import { VALIDATE_EMAIL } from "../graphql/queries";
+import { useLocalStorage } from "../hooks/useLocalStorage";
+import { EmailVars, Response } from "../interfaces/User";
+import { GlobalStyle } from "../theme";
+import { REPLACE, STORAGE_KEY } from "../utils/constants";
+import { Button, Form, H1, Input, ValidationText } from "./styled";
+
+function isValidEmail(email: string) {
+  const regex =
+    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return regex.test(email);
+}
+
+export const LoginForm = () => {
+  const { loggedUser, setLoggedUser } = useContext(UserContext);
+  const [errorMessage, setErrorMessage] = useState("");
+  const [emailInput, setEmailInput] = useState("");
+  const [, setUserInStorage] = useLocalStorage(STORAGE_KEY.USER, "");
+  const [validateEmail, { loading, error, data }] = useLazyQuery<
+    Response,
+    EmailVars
+  >(VALIDATE_EMAIL);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (loggedUser) navigate("/", REPLACE);
+    if (!data) return;
+    if (!data?.users.length) {
```

Comment:
why don't we use a single condition for line 33 and 34?

Reply:
I recently fixed this, it was done that way because when I coded this component I had logic when the 2nd conditional passed, but now they are kind of the same